sys/src/9/amd64/dat.h

```diff
@@ -354,69 +351,88 @@ struct Stackframe
 };
 
 /*
- * This is the low memory map, between 0x100000 and 0x110000.
- * It is located there to allow fundamental datastructures to be
+ * This is the low memory map, between 1MiB and 2MiB.
+ *
+ * It is located there to allow fundamental data structures to be
  * created and used before knowing where free memory begins
  * (e.g. there may be modules located after the kernel BSS end).
- * The layout is known in the bootstrap code in l32p.s.
+ * The layout is known in the bootstrap code in entry.S
+ *
  * It is logically two parts: the per processor data structures
  * for the bootstrap processor (stack, Mach, vsvm, and page tables),
  * and the global information about the system (syspage, ptrpage).
  * Some of the elements must be aligned on page boundaries, hence
  * the unions.
  */
 struct Sys {
-	unsigned char	machstk[MACHSTKSZ];
+	alignas(4096) unsigned char machstk[MACHSTKSZ];
 
-	PTE	pml4[PTSZ/sizeof(PTE)];	/*  */
-	PTE	pdp[PTSZ/sizeof(PTE)];
-	PTE	pd[PTSZ/sizeof(PTE)];
-	PTE	pt[PTSZ/sizeof(PTE)];
+	PTE	ipml4[PTSZ/sizeof(PTE)];	// Only used very early in boot
```

Comment:
is PTSZ / sizeof(PTE) worth a define at some point?

Reply:
Probably, but I think best deferred to a followup change.